src/integration-test/java/edu/kit/ipd/crowdcontrol/objectservice/mail/MailHandlerTest.java

```diff
@@ -75,5 +77,79 @@ public void test() throws Exception {
             }
         }
         assertTrue(found);
+
+        messages = fetcher.fetchFolder(folder);
+        found = false;
+        for (Message message : messages) {
+            if (message.getSubject().equals(subject)) {
+                found = true;
+                break;
+            }
+        }
+        assertFalse(found);
+    }
+
+    @Test
+    public void testFetchUnseenMarkAsUnseen() throws Exception {
+        String uuid = UUID.randomUUID().toString();
+        String subject = "[test] " + uuid;
+
+        sender.sendMail(receiver, subject, uuid);
+        try {
+            Thread.sleep(3000);
+        } catch (InterruptedException e) {
+        }
+        Message[] messages = fetcher.fetchUnseen(folder);
+
+        boolean found = false;
+
+        Message testMessage = null;
+        for (Message message : messages) {
+            if (message.getSubject().equals(subject)) {
+                found = true;
+                testMessage = message;
+                break;
+            }
+        }
+        assertTrue(found);
+
+        messages = fetcher.fetchUnseen(folder);
+
+        found = false;
+        for (Message message : messages) {
+            if (message.getSubject().equals(subject)) {
+                found = true;
+                break;
+            }
+        }
+        assertFalse(found);
```

Comment:
Does fetching Mark as seen automatically or do you do that? You might want to do this only after you processed the fetched message, so it doesn't get lost if we crash in the middle.